frappe/model/base_document.py

```diff
@@ -371,6 +371,13 @@ def get_valid_dict(
 				elif df.fieldtype in float_like_fields and not isinstance(value, float):
 					value = flt(value)
 
+				elif df.fieldtype == "Rating":
```

Comment:
It's best to move this to document's `_validate` method. `get_valid_dict` is serializer function, so it shouldn't be applying business logic here.